### DISCUSSION TITLE
Add search functionality

### DISCUSSION THREAD

**Discussion by DeGraciaMathieu** [Halleck45/ast-metrics#29] [Ideas]
Hello @Halleck45 

Returning to our [discussion](https://github.com/Halleck45/ast-metrics/issues/1#issuecomment-2016532845). 

> Regarding the search functionality, what do you think? Should we add a --filter argument?

For the executable "ast-metrics'", i don't think it's necessary, simply specify several paths.

> maybe a search field that appears in the window when you press Ctrl + F?

Yes, definitely a good idea.

In my analysis tools, I typically enable pattern-based searches. For example, entering "command" would locate all classes with that suffix, such as "UserCommand".

**Comment by Halleck45** [comment]
Hi @DeGraciaMathieu 

I think this idea is implemented in https://github.com/Halleck45/ast-metrics/pull/32

[capture-ast.webm](https://github.com/Halleck45/ast-metrics/assets/1076296/c6b3bb55-5df1-4a54-bcbb-0c30abcaf719)

Does that sound good?

**Reply by DeGraciaMathieu** [reply]
Hi,

Yes, that sounds good and functional !

Just one point : is it possible to make the size of the class name more dynamic ? With complex applications, this quickly becomes unreadable.

![image](https://github.com/Halleck45/ast-metrics/assets/11473997/acc3cafc-420c-4711-934b-e7a00bc497f4)

**Reply by Halleck45** [reply]
I'm closing the discussion since the main topic is resolved. I'm lacking some ideas for the full name. Animating on hover is quite complex and will look weird. I can truncate on the left otherwise. Or, I can display the full name of the current line in a "status bar", for example.

The last idea seems to me the most ergonomic.
